authenticity.cpp

```diff
@@ -189,37 +185,30 @@ std::string Authenticity::Assay
         }
 
     // Validate all data files.
-    fs::path original_path(fs::current_path());
-    if(0 != chdir(data_path.string().c_str()))
+    std::vector<md5sum_for_file> sums;
+    std::string error_message, md5;
+    if (!md5sum_vec_from_file((data_path / md5sum_file()).string(), sums, error_message))
         {
         oss
-            << "Unable to change directory to '"
-            << data_path
-            << "'. Try reinstalling."
+            << "Failed to read md5sum file. Try reinstalling." << std::endl
+            << "Error: " << error_message
             ;
         return oss.str();
         }
-    try
+    for (const auto& s : sums)
```

Comment:
```suggestion
    for (auto const& s : sums)    
```
Please use "east const" consistently in lmi sources.